File: library/src/main/java/com/github/javiersantos/piracychecker/enums/InstallerID.java
package com.github.javiersantos.piracychecker.enums;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collections;
import java.util.List;

public enum InstallerID {
    GOOGLE_PLAY("com.android.vending|com.google.android.feedback"),
    AMAZON_APP_STORE("com.amazon.venezia"),
    GALAXY_APPS("com.sec.android.app.samsungapps");

    private final String text;

    InstallerID(final String text) {
        this.text = text;
    }

    /* (non-Javadoc)
     * @see java.lang.Enum#toString()
     */
    @Override
    public String toString() {
        return text;
    }

    public List<String> toIDs() {
        if (text.contains("|")) {
            String[] split = text.split("\\|");
            return new ArrayList<>(Arrays.asList(split));
        } else {
            return new ArrayList<>(Collections.singletonList(text));
        }
    }

}

File: library/src/main/java/com/github/javiersantos/piracychecker/enums/InstallerID.kt
package com.github.javiersantos.piracychecker.enums

import java.util.ArrayList
import java.util.Arrays

enum class InstallerID(private val text: String) {
    GOOGLE_PLAY("com.android.vending|com.google.android.feedback"),
    AMAZON_APP_STORE("com.amazon.venezia"),
    GALAXY_APPS("com.sec.android.app.samsungapps");
    
    /* (non-Javadoc)
     * @see java.lang.Enum#toString()
     */
    override fun toString(): String {
        return text
    }
    
    fun toIDs(): List<String> = if (text.contains("|")) {
        val split = text.split("\\|".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
        ArrayList(Arrays.asList(*split))
    } else {
        ArrayList(listOf(text))
    }
}
